internal/datastore/common/url.go

```diff
@@ -0,0 +1,19 @@
+package common
+
+import (
+	"errors"
+	"net/url"
+)
+
+// MetricsIDFromURL extracts the metrics ID from a given datastore URL.
+func MetricsIDFromURL(dsURL string) (string, error) {
+	if dsURL == "" {
+		return "", errors.New("datastore URL is empty")
+	}
+
+	u, err := url.Parse(dsURL)
+	if err != nil {
+		return "", errors.New("could not parse datastore URL")
+	}
+	return u.Host + u.Path, nil
```

Comment:
note that returning host+path will lead to explosion of cardinality when having many spiced clusters in the same physical database. I do see the value in disambiguating telemetry across spicedb clusters, just something to note.